src/transport/SecurePairingSession.cpp

```diff
@@ -450,13 +425,14 @@ CHIP_ERROR SecurePairingSession::HandleCompute_cA(const PacketHeader & header, S
     return err;
 }
 
-CHIP_ERROR SecurePairingSession::HandlePeerMessage(const PacketHeader & packetHeader, System::PacketBuffer * msg)
+CHIP_ERROR SecurePairingSession::HandlePeerMessage(const PacketHeader & packetHeader, System::PacketBuffer * msgIn)
 {
     CHIP_ERROR err      = CHIP_NO_ERROR;
     uint16_t headerSize = 0;
     PayloadHeader payloadHeader;
+    System::AutoFreePacketBuffer msg(msgIn);
```

Comment:
I assumed a bug .. let me check. Not freeing in special cases makes no sense: all our functions either always free or always not free.

Reply:
I believe we have just buggy code.

We still have it buggy: `CHIP_ERROR RendezvousSession::HandlePairingMessage` seems to never free on error.

Generally I believe this 'free or not free' is hitting us hard. We really need the full fix.

Reply:
I think 'release only on success' is a bug: all our other functions either 'release always' or 'release never'.
Release only on success sounds very hard to account for.

This will be fixed in PRs by kevin.

Reply:
Works for me.